include/sys/cbprintf.h

```diff
@@ -21,6 +21,21 @@
 extern "C" {
 #endif
 
+/* Determine if _Generic is supported.
+ * In general it's a C11 feature but it was added also in:
+ * - GCC 4.9.0 https://gcc.gnu.org/gcc-4.9/changes.html
+ * - Clang 3.0 https://releases.llvm.org/3.0/docs/ClangReleaseNotes.html
+ */
+#ifndef __C_GENERIC
+#if ((__STDC_VERSION__ >= 201112L) || \
```

Comment:
It might be better to integrate this into the introduction of the API.

Also `__C_GENERIC` is not a legal identifier for anybody but the toolchain to use.  Even if somebody wants to say "but we *are* the system so we can do whatever we want" it'll confuse people.  `Z_C_GENERIC` maybe.

Reply:
> It might be better to integrate this into the introduction of the API.

do you mean squashing commits?